File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> ans;
        if(!root) return ans;
        queue<TreeNode*> q;
        q.push(root);
        TreeNode* front= NULL;
        while(!q.empty()){
            int sz=q.size();
            vector<int> currLvL;
            while(sz--){
                front = q.front();
                q.pop();
                currLvL.push_back(front->val);
                if(front->left) q.push(front->left);
                if(front->right) q.push(front->right);
            }
            ans.push_back(currLvL);
        }
        return ans;
    }
};

File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> levelOrderList = new ArrayList<>();
        Queue<TreeNode> q = new LinkedList<>();
        if(root==null) return levelOrderList;
        q.add(root);
        while(!q.isEmpty()){
            int siz = q.size();
            List<Integer> temp = new ArrayList<>();
            while(siz-->0){
                TreeNode rem = q.remove();
                temp.add(rem.val);
                if(rem.left!=null){
                    q.add(rem.left);
                }
                if(rem.right!=null){
                    q.add(rem.right);
                }
            }
            levelOrderList.add(temp);
        }
        return levelOrderList;
    }
}